src/crawlee/memory_storage_client/_request_queue_client.py

```diff
@@ -240,39 +237,39 @@ async def add_request(
         if existing_queue_by_id is None:
             raise_on_non_existing_storage(StorageTypes.REQUEST_QUEUE, self.id)
 
-        request_model = await self._create_internal_request(request, forefront)
+        internal_request = await self._create_internal_request(request, forefront)
 
         async with existing_queue_by_id.file_operation_lock:
-            existing_request_with_id = existing_queue_by_id.requests.get(request_model.id)
+            existing_request_with_id = existing_queue_by_id.requests.get(internal_request.id)
 
             # We already have the request present, so we return information about it
             if existing_request_with_id is not None:
                 await existing_queue_by_id.update_timestamps(has_been_modified=False)
 
                 return ProcessedRequest(
-                    id=request_model.id,
-                    unique_key=request_model.unique_key,
+                    id=internal_request.id,
+                    unique_key=internal_request.unique_key,
                     was_already_present=True,
-                    was_already_handled=existing_request_with_id.order_no is None,
+                    was_already_handled=existing_request_with_id.handled_at,
```

Comment:
`handled_at` is `datetime` or `None`, but `was_already_handled` is boolean. And why this is not catched by type checker?

Reply:
I wish I could say that after adding the type annotation the type checker started pointing out an error. But it just ignores it, and it's a mystery to me why.